rust/src/interpreter.rs

```diff
@@ -408,29 +409,35 @@ impl<'a> Interpreter<'a> {
                     break;
                 }
                 Err(GetOpcodeError::Invalid) => {
-                    return Err(FailStatus::InvalidInstruction);
+                    return FailStatus::InvalidInstruction.into();
```

Comment:
I am not sure if this is more idiomatic than `Err(FailStatus::InvalidInstruction)` why do you prefer this?

Reply:
Because the return type is no longer Result<(), FailStatus> but now either ExecutionResult or StepResult, Err(FailStatus::InvalidInstruction) is no longer valid. FailStatus::InvalidInstruction.into() converts the fail status into one of the two result structs.